code/WorkInProgress/nadir_antenna.dm

```diff
@@ -690,8 +690,7 @@ TYPEINFO(/obj/machinery/transception_pad)
 						src.attempt_transceive()
 					if("receive")
 						var/sigindex = signal.data["data"]
-						if(isnum_safe(sigindex))
-							src.attempt_transceive(sigindex)
+						src.attempt_transceive(text2num_safe(sigindex))
```

Comment:
Run text2num_safe on the signal.data and leave the isnum_safe check in to account for null/invalid data. If a null is passed into `attempt_transceive` it will try to send instead of receiving.